runtime/src/bpftime_prog.cpp

```diff
@@ -26,7 +27,18 @@ bpftime_prog::bpftime_prog(const struct ebpf_inst *insn, size_t insn_cnt,
 {
 	SPDLOG_DEBUG("Creating bpftime_prog with name {}", name);
 	insns.assign(insn, insn + insn_cnt);
-	vm = ebpf_create();
+	const char *vm_name = bpftime::bpftime_get_agent_config().get_vm_name();
```

Comment:
I suggest passing an reference of agent_config when constructing bpftime_prog. That argument could be  optional, if not provided, we may use the one from global shared memory